### ISSUE TITLE
V12 Scene Region limitations

### ISSUE THREAD

**Issue by Plushtoast** [foundryvtt/foundryvtt#11108]
### User Experience

Hi,

I've got some feedback regarding blocking points on extending scene regions. Please refer to this [Discord link](https://discord.com/channels/170995199584108546/1184176344276406292/1247103830533865533) for additional context.

## Missing Hooks

Currently, there are no hooks connected (other than for app rendering), which limits the ability to modify core, system or module-specific behavior.

I propose the following solutions:

* Add a new `CallHookBehavior` that simply calls a hook previous to all other events to modify or prevent specific other behaviors, or
* Add at least a hook maybe around here?:
![grafik](https://github.com/foundryvtt/foundryvtt/assets/44941845/50269a4f-417c-4ade-aad0-86e0c6d5980c)

Even better would be behavior-specific hooks.

## Private Methods

Many new features in v12 are locked behind private methods, preventing modification or inheritance. This often necessitates duplicating code to add functionality.

A notable example is the `TokenTeleportBehavior`, which is almost completely inaccessible. Why is this? While private methods for important core functionality are understandable, this seems overly cautious for minor features.

## Teleport Token Behavior

The current behavior randomly places the token in the destination area. This works for teleporters like portals but not for broad stairs, ledges, and similar structures.

The teleport behavior should allow the token to appear in the destination area at the same relative position as where it entered the source area. For instance, if a rooftop side is 10 grid squares long and you can jump anywhere, the current behavior would always place the token in a random spot, regardless of whether you jumped off the bottom-left or bottom-right corner of the roof.

I would prefer that if you enter the source area 25% from the left and 25% from the top, you should also exit the target area 25% from the left and 25% from the top. For rectangles and circles, this makes sense. For polygons, it could check if the point exists, and if not, place it randomly. 

Overall, the current teleport behavior isn't a worthy replacement for the outdated multilevel-token module and is unnecessarily difficult to modify.

I am aware of the macro behavior, but this is just an inconvenient workaround, which would discourage replacing multilevel-tokens.

**Comment by aaclayton** [comment]
Please split this into 3 issues so we can evaluate each independently. Once that is done I'll mark this as unblocked, otherwise this is going nowhere.

**Comment by Plushtoast** [comment]
Moved  to:
#11115
#11116 
#11118

**Comment by aaclayton** [comment]
Thank you!
